### ISSUE TITLE
bolt: decode command buffer fails with GPU OOM and cb.error is never checked → unbounded token-id-0 (`!`) output above ~2–6K prompt tokens

### ISSUE THREAD

**Issue by penta2himajin** [penta2himajin/qwisp#169]
## Summary

On the bolt (streaming default) tier, prompts above roughly 2–6K tokens produce a single correct
first token followed by an unbounded run of `!`. The `!` is **token id 0**.

The cause is not model quality and not a numeric overflow. **The decode command buffer fails with
a GPU out-of-memory error, and nothing in the shipped path checks `cb.error`.** The GPU writes
nothing, the logits buffer stays at its zero-initialised state, and `argmax_rows` tie-breaks a
constant row to index 0 — token id 0, forever.

## Evidence

Instrumented `SeedlessFusedForward.chainedStepArgmax` (the path bolt decode actually uses) right
after `cb.waitUntilCompleted()`. Same machine, same config (`QWISP_DEVICE_RAM=16`,
`sysctl iogpu.wired_limit_mb=10900`, mixed ON = default, loop guard OFF, `--max-tokens 8`),
one natural-prose prompt truncated to two lengths:

| prompt | output | `cb.status` | `cb.error` |
|---|---|---|---|
| 1,770 tok | `Here are the five most important decisions and` | 4 (completed) | nil |
| 6,559 tok | `Here!!!!!!!` | **5 (error)** | **`MTLCommandBufferErrorDomain Code=8 "Insufficient Memory (00000008:kIOGPUCommandBufferCallbackErrorOutOfMemory)"`** |

Buffer contents at the same instant:

| buffer | 1,770 tok | 6,559 tok |
|---|---|---|
| `hBuf` (pre-final-norm hidden, 2048) | min −2.25 max 2.38, nan 0, zero 0 | min −1.46 max 2.76, nan 0, zero 0 |
| `normed` (lm_head input, 2048) | min −20.14 max 17.64, nan 0, zero 0 | min −17.95 max 27.77, nan 0, zero 0 |
| `logits` (V = 248,320) | min −10.11 max 25.17, nan 0, **zero 0** | min 0.0 max 0.0, nan 0, **zero 248,320 (all of V)** |

`hBuf` and `normed` look healthy in the failing case because they hold leftover values from
prefill — the decode CB never ran, so nothing updated them. `logits` is `.storageModeShared` and
zero-initialised at allocation, so "never written" and "all zeros" are the same state.

Emitted token ids (`QWISP_TOK_DUMP`): `8160` (= `Here`) then `0` repeated. `vocab.json` id 0 = `!`.

## Why this explains every prior observation

- **Length dependence.** KV grows with the prompt, headroom shrinks; 1.8K fits, 6.6K does not.
- **Decode reported at 1832 tok/s** (guard off). The GPU is doing nothing.
- **`QWISP_MIXED=0` fixes it** (clean output, 68.8 tok/s). Mixed residency at C=128 resolves to
  `K4=0 + M2=256` — coverage 256/layer, i.e. every expert resident. Generic bolt keeps 128.
  Halving resident experts brings the budget back under the limit.
- **strict is healthy at 41K** (correct summary, 20.3 tok/s). Its budget-fit picks C=64.
- **The loop guard cannot help.** After rewinding to strict it re-prefills prompt + the emitted
  `!`s and bit-exactly continues them — that is the correct continuation of that context.
- Earlier hypotheses are refuted by the same data: **not NaN** (`nan=0`; also, `argmax_rows`
  initialises `bi = 0x7fffffff`, so an all-NaN row would emit 2147483647, not 0), and **not a
  2-bit quality wall** (a quality wall yields finite, skewed logits — these are exactly zero;
  and with coverage 256 every expert is resident regardless of prompt length, so a static
  precision limit cannot be length-dependent).

## Repro

```bash
sudo sysctl iogpu.wired_limit_mb=10900       # 16GB-class emulation; undo with =0
env QWISP_MODEL=<model> QWISP_DEVICE_RAM=16 QWISP_BOLT_STABILITY_GUARD=0 \
    qwisp chat --no-thinking --max-tokens 8 "$(head -c 24300 <long-prose-file>)"
```

Expected: a coherent answer. Actual: `Here!!!!!!!`.

## Proposed fix, in order

1. **Check `cb.status` / `cb.error` on every committed command buffer in the decode and verify
   paths, and refuse to emit tokens from a failed buffer.** A silent GPU OOM that produces
   plausible-looking output is the worst available failure mode: the user gets garbage with no
   signal, and every downstream heuristic (loop guard, benchtest) reads it as model behaviour.
   This is tier-independent — the same silent failure is reachable on 8GB.
2. **Do not enable mixed residency at C=128.** Mixed was validated at the 8GB byte budget where
   it buys coverage 128; at C=128 it resolves to coverage 256 and doubles resident expert bytes.
   Recorded per-layer footprint is ~103 experts, so generic bolt at C=128 already exceeds it —
   there is no capacity deficit for mixed to fix at 16GB.
3. **Budget-fit does not account for mixed's real residency.** The sizing path picks C from the
   byte budget without knowing that mixed will resolve to 2× the slot count. Related: #164.

## Scope

Anyone on the streaming/bolt default with a prompt past the threshold. #86 is in this range
today; `--lossless` is a working workaround right now (slower, but correct).

The threshold between 1,770 and 6,559 prompt tokens is being bisected next — with `cb.error`
visible it can be read directly rather than inferred from output shape.

## Notes for whoever picks this up

Bolt's decode tokens come from `SeedlessFusedForward.chainedStepArgmax`
(`swift/Sources/QwispCore/SeedlessFusedVerify.swift`), **not** `stepArgmax`. `Tell.SpecBackend`
is a struct of closures whose `stepArgmax` is a `var` that other features swap out, and
`streamingBackend` falls back to an `MLX.argMax`-composed step when `fwd.head == nil`. Instrument
all three or you will measure the wrong path — MLX's argMax returns index 0 on an all-NaN row,
which makes it easy to "confirm" a NaN hypothesis that is not what is happening.
